src/tests/ftest/pool/label.py

```diff
@@ -0,0 +1,207 @@
+                # Failed for the wrong reason.
+                error_message = "dmg pool create failed for the wrong " +\
+                    "reason! Expected to exist = {}".format(expected_error)
+                errors.append(error_message)
+
+        except TestFail as error:
+            errors.append(error)
+            self.log.info("dmg failed!")
+
+        finally:
+            self.pool[-1].dmg.exit_status_exception = True
+
+        return errors
+
+    @skipForTicket("DAOS-8183")
+    def test_valid_labels(self):
+        """Test ID: DAOS-7942
+
+        Test Description: Create and destroy pool with the following labels.
+        * Random alpha numeric string of length 126.
+        * Random alpha numeric string of length 127.
+        * Random alpha numeric string of length 128.
+        * Random upper case string of length 50.
+        * Random lower case string of length 50.
+        * Random number string of length 50.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=small
+        :avocado: tags=pool,create_valid_labels
+        """
+        self.pool = []
+        errors = []
+        labels = [
+            get_random_string(126),
+            get_random_string(127),
+            get_random_string(128),
+            get_random_string(length=50, include=string.ascii_uppercase),
+            get_random_string(length=50, include=string.ascii_lowercase),
+            get_random_string(length=50, include=string.digits)
+        ]
+
+        for label in labels:
+            errors.extend(self.verify_create(label, False))
+            errors.extend(self.verify_destroy(self.pool[-1], False))
+
+        report_errors(self, errors)
+
+    def test_invalid_labels(self):
+        """Test ID: DAOS-7942
+
+        Test Description: Create pool with following invalid labels.
+        * UUID format string: 23ab123e-5296-4f95-be14-641de40b4d5a
+        * Long label - 129 random chars.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=small
+        :avocado: tags=pool,create_invalid_labels
+        """
+        self.pool = []
+        errors = []
+        label_outs = [
+            ("23ab123e-5296-4f95-be14-641de40b4d5a", "invalid label"),
+            (get_random_string(129), "value too long")
+        ]
+
+        for label_out in label_outs:
+            errors.extend(self.verify_create(label_out[0], True, label_out[1]))
+
+        report_errors(self, errors)
+
+    def test_duplicate_create(self):
+        """Test ID: DAOS-7942
+
+        Test Description:
+        1. Create a pool with a label.
+        2. Create another pool with the same label. Should fail.
+        3. Destroy the pool.
+        4. Create a pool with the same label again. It should work this time.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=small
+        :avocado: tags=pool,duplicate_label_create
+        """
+        self.pool = []
+        label = "TestLabel"
```

Comment:
Option to use random labels:
```suggestion
        label = get_random_string(random.choice(range(124) + 1))
```